Disable judgments when timeline is disabled

duartegalvao opened the pull request:
This PR disables all editing actions on the timeline when it is detected that it became outdated, avoiding errors due to editing actions on the wrong revision.
![image](https://github.com/user-attachments/assets/85f1e23a-229b-4fb7-a691-6e86edc5dbcd)

ThiefMaster commented on indico/modules/events/editing/client/js/editing/timeline/index.jsx:
```diff
@@ -46,10 +46,6 @@ export default function Timeline() {
     };
   }, [dispatch]);
 
-  const refresh = () => {
-    dispatch(actions.useUpdatedTimeline());
-  };
```
Just out of curiosity, what was the reason for inlining this? Don't you find it more readable to have short functions in the JSX and the actual logic above?

ThiefMaster commented:
![image](https://github.com/user-attachments/assets/94c10da3-477b-4908-b381-a031cb6fe0bb)

shouldn't this also be disabled?

duartegalvao commented:
> ![image](https://private-user-images.githubusercontent.com/179599/356287020-94c10da3-477b-4908-b381-a031cb6fe0bb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMxODcxMTYsIm5iZiI6MTcyMzE4NjgxNiwicGF0aCI6Ii8xNzk1OTkvMzU2Mjg3MDIwLTk0YzEwZGEzLTQ3N2ItNDkwOC1iMzgxLWEwMzFjYjZmZTBiYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgwOVQwNzAwMTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hODZlOGU1MDkwMTM0NmE5ZDZiNTBjNDI3NGI0N2RjOTk0YTZlZTA0NGMzMjlkOGFjMjBlOWQ5MmRmMWFhMmJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.hdyfuPKJkPJNui4EIdFlIPAZdF0OCUNVpOyCJZwuzq4)
> 
> shouldn't this also be disabled? (saving it failed of course, since it was outdated because i reverted a revision in the other tab)



> ![image](https://private-user-images.githubusercontent.com/179599/356287020-94c10da3-477b-4908-b381-a031cb6fe0bb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMxODcxMTYsIm5iZiI6MTcyMzE4NjgxNiwicGF0aCI6Ii8xNzk1OTkvMzU2Mjg3MDIwLTk0YzEwZGEzLTQ3N2ItNDkwOC1iMzgxLWEwMzFjYjZmZTBiYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgwOVQwNzAwMTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hODZlOGU1MDkwMTM0NmE5ZDZiNTBjNDI3NGI0N2RjOTk0YTZlZTA0NGMzMjlkOGFjMjBlOWQ5MmRmMWFhMmJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.hdyfuPKJkPJNui4EIdFlIPAZdF0OCUNVpOyCJZwuzq4)
> 
> shouldn't this also be disabled? (saving it failed of course, since it was outdated because i reverted a revision in the other tab)

indeed it should, I'll fix it

duartegalvao commented on indico/modules/events/editing/client/js/editing/timeline/ReviewForm.jsx:
```diff
@@ -106,43 +99,53 @@ export default function ReviewForm() {
+    <Popup
+      trigger={
+        <div className="flexrow f-a-center" styleName="judgment-form">
+          <CommentForm onSubmit={createComment} onToggleExpand={setCommentFormVisible} />
```
I can't find a way to pass a `disabled` prop to the `CommentForm` without resetting its state...

ThiefMaster replied:
Isn't this what you need here? (didn't bother with proptypes to test this ;))

```patch
diff --git a/indico/modules/events/editing/client/js/editing/timeline/CommentForm.jsx b/indico/modules/events/editing/client/js/editing/timeline/CommentForm.jsx
index 2378acd0ba..0da6e66771 100644
--- a/indico/modules/events/editing/client/js/editing/timeline/CommentForm.jsx
+++ b/indico/modules/events/editing/client/js/editing/timeline/CommentForm.jsx
@@ -18,7 +18,7 @@ import {getDetails} from './selectors';

 import './CommentForm.module.scss';

-export default function CommentForm({onSubmit, onToggleExpand, initialValues, expanded}) {
+export default function CommentForm({onSubmit, onToggleExpand, initialValues, expanded, disabled}) {
   const {canCreateInternalComments} = useSelector(getDetails);
   const [commentFormVisible, setCommentFormVisible] = useState(expanded);

@@ -67,7 +67,7 @@ export default function CommentForm({onSubmit, onToggleExpand, initialValues, ex
                   />
                 )}
                 <Form.Group styleName="submit-buttons" inline>
-                  <FinalSubmitButton label={Translate.string('Comment')} />
+                  <FinalSubmitButton label={Translate.string('Comment')} disabled={disabled} />
                   <Button
                     disabled={fprops.submitting}
                     content={Translate.string('Cancel')}
diff --git a/indico/modules/events/editing/client/js/editing/timeline/ReviewForm.jsx b/indico/modules/events/editing/client/js/editing/timeline/ReviewForm.jsx
index 0817eddea6..2be19abfe6 100644
--- a/indico/modules/events/editing/client/js/editing/timeline/ReviewForm.jsx
+++ b/indico/modules/events/editing/client/js/editing/timeline/ReviewForm.jsx
@@ -102,7 +102,11 @@ export default function ReviewForm() {
     <Popup
       trigger={
         <div className="flexrow f-a-center" styleName="judgment-form">
-          <CommentForm onSubmit={createComment} onToggleExpand={setCommentFormVisible} />
+          <CommentForm
+            onSubmit={createComment}
+            onToggleExpand={setCommentFormVisible}
+            disabled={isOutdated}
+          />
           {canPerformSubmitterActions && canReview && !editor && (
             <>
               <span className="comment-or-review">
```

duartegalvao replied:
thats what I did, at least for me this clears the form when the timeline becomes outdated. not sure why
